src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/OveruseEstimator.java

```diff
@@ -229,14 +232,18 @@ public void update(
 
     private double updateMinFramePeriod(double tsDelta)
     {
-        double minFramePeriod = tsDelta;
-
-        if (tsDeltaHist.size() >= kMinFramePeriodHistoryLength)
-            tsDeltaHist.remove(0);
-        for (Double d : tsDeltaHist)
-            minFramePeriod = Math.min(d, minFramePeriod);
-        tsDeltaHist.add(tsDelta);
-        return minFramePeriod;
+        tsDeltaHist[tsDeltaHistIdx] = tsDelta;
+        tsDeltaHistIdx = (tsDeltaHistIdx + 1) % tsDeltaHist.length;
```

Comment:
Is that (much) faster than the following? I'm curious if you have the answer (i.e. I don't want you to research it or change your source code).
```java
int nextTsDeltaHistIdx = tsDeltaHistIdx + 1;
tsDeltaHistIdx = (nextTsDeltaHistIdx < tsDeltaHist.length) ? nextTsDeltaHistIdx : 0;
```